parser/ParseBlockProperties.hpp

```diff
@@ -0,0 +1,247 @@
+/**
+ *   Copyright 2016 Pivotal Software, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#ifndef QUICKSTEP_PARSER_PARSE_BLOCK_PROPERTIES_HPP_
+#define QUICKSTEP_PARSER_PARSE_BLOCK_PROPERTIES_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "parser/ParseString.hpp"
+#include "parser/ParseTreeNode.hpp"
+#include "utility/Macros.hpp"
+#include "utility/PtrList.hpp"
+
+namespace quickstep {
+
+/** \addtogroup Parser
+ *  @{
+ */
+
+/**
+ * @brief Parsed BlockPropertyItem which simply a key (property) and the 
+ *  corresponding value.
+ * @details The value may actually be a list of Parse strings. This is to cover 
+ *  the case where a user specifies a subset of columns to compress rather than 
+ *  using the keyword 'ALL'. 
+ */
+class ParseBlockPropertyItem : public ParseTreeNode {
+ public:
+  enum class Property {
+    kCompress,
+    kSort,
+    kType
+  };
+
+  /**
+   * @brief Constructor.
+   *
+   * @param line_number Line number of the first token of this node in the SQL 
+   *  statement.
+   * @param column_number Column number of the first token of this node in the 
+   *  SQL statement.
+   * @param property Describes which type of block property this is. Corresponds
+   *  to a field of the StorageBlockLayoutDescription message.
+   * @param value A parse string representing the assignment of the property. 
+   *  ParseBlockPropertyItem takes ownership of the pointer.
+   **/
+  ParseBlockPropertyItem(int line_number,
+                         int column_number,
+                         Property property,
+                         ParseString* value)
+                             : ParseTreeNode(line_number, column_number),
```

Comment:
This is not the correct way to indent. Please read the similar files to get familiar with the code style.

```
  ParseBlockPropertyItem(int line_number, typeX xxx)
      : ParseTreeNode(line_number),
         xxx_(xxx) {
  }
```